packages/containers/src/EditableText/EditableText.container.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { EditableText as Component } from '@talend/react-components';
+import Immutable from 'immutable';
+import omit from 'lodash/omit';
+import { cmfConnect } from '@talend/react-cmf';
+
+export const DISPLAY_NAME = 'Container(EditableText)';
+export const DEFAULT_STATE = new Immutable.Map({
+	editMode: false,
+});
+
+class EditableText extends React.Component {
+	static displayName = DISPLAY_NAME;
+
+	static propTypes = {
+		...cmfConnect.propTypes,
+		actionCreatorCancel: PropTypes.string,
+		actionCreatorEdit: PropTypes.string,
+		actionCreatorSubmit: PropTypes.string,
+		actionCreatorChange: PropTypes.string,
+		actionCreatorGoBack: PropTypes.string,
+		onCancel: PropTypes.func,
+		onEdit: PropTypes.func,
+		onSubmit: PropTypes.func,
+		onChange: PropTypes.func,
+		text: PropTypes.string,
+	};
+
+	static contextTypes = {
+		registry: PropTypes.object,
+		store: PropTypes.object,
+	};
+
+	constructor(props) {
+		super(props);
+		this.onCancel = this.onCancel.bind(this);
+		this.onEdit = this.onEdit.bind(this);
+		this.onSubmit = this.onSubmit.bind(this);
+		this.onChange = this.onChange.bind(this);
+	}
+
+	onSubmit(event, data) {
+		this.props.setState(() => ({
+			editMode: false,
+		}));
+		if (this.props.onSubmit) {
+			this.props.onSubmit(event, data);
+		}
+		if (this.props.actionCreatorSubmit) {
+			this.props.dispatchActionCreator(this.props.actionCreatorSubmit, event, {
+				props: this.props,
+				data,
+			});
+		}
+	}
+
+	onCancel(event) {
+		this.props.setState(() => ({
+			editMode: false,
+		}));
+		if (this.props.onCancel) {
+			this.props.onCancel(event);
+		}
+		if (this.props.actionCreatorCancel) {
+			this.props.dispatchActionCreator(this.props.actionCreatorCancel, event, {
+				props: this.props,
+			});
+		}
+	}
+
+	onEdit(event) {
+		this.props.setState(() => ({
+			editMode: !this.props.state.get('editMode', false),
```

Comment:
Same here, or take the state from the arguments instead of `props.state`

Reply:
BTW this seems weird in fact, don't you want to set `editMode` to true instead of toggle it ?

Reply:
oh you're probably right. I moved this code from the SubHeader container, but I agree, I don't see any reason to toggle it here. I corrected it, thanks.